### ISSUE TITLE
coreutils cp does not resolve relative symlinks against the correct base directory

### ISSUE THREAD

**Issue by kevinburke** [uutils/coreutils#2631]
I'm trying to compile Postgres. I am trying to run the following command:

```
cp ./utils/probes.h /tmp
```

`probes.h` is a symlink:

```
lrwxr-xr-x  1 kevin   35 Sep  2 21:45 probes.h -> ../../../src/backend/utils/probes.h
```

GNU `cp` correctly resolves this symlink, if I run this from either inside the "utils" directory, or outside of it.

Coreutils `cp` only resolves this symlink if it's run from inside the containing "utils" directory ie.

```
cp probes.h /tmp
```

If I run it from up one level, I get the following error:

```
$ cp ./utils/probes.h /tmp
/Users/kevin/local/coreutils/bin/cp: '../../../src/backend/utils/probes.h' -> '/tmp/probes.h': No such file or directory (os error 2)
```

I believe that Rust coreutils is trying to resolve the symlink relative to the working directory, when it needs to resolve the symlink relative to the directory containing the link.

**Comment by sylvestre** [comment]
you tried with main of coreutils, right ?

**Comment by miDeb** [comment]
Thanks for the bug report! This will be fixed by #2624.
